### PR TITLE
Task/sapnamysore/tlt 2119/jasmine confirm add people to course

### PR DESCRIPTION
**Pull request by sapnamysore**
-Made the modalInstance a scope variable in order to close and dismiss the dialog for Jasmine testing(This is similar to how is set in confirmRemove). Also tested the change locally.
-Jasmine tests

### CODE REVIEW

File: course_info/static/course_info/js/controllers/PeopleController.js [hunk 1]
```diff
@@ -198,7 +198,7 @@
             var searchTermList = $scope.getSearchTermList(searchTerms);
             // open a modal confirmation box and as the user to verify they want to add
             // the number of users they entered.
-            var modalInstance = $uibModal.open({
+            $scope.modalInstance = $uibModal.open({
```
Comment (elliottyates) [inline course_info/static/course_info/js/controllers/PeopleController.js]: If we're going to put this on the scope, can we call it something more specific, like `confirmAddModal`?

File: course_info/static/course_info/js/controllers/PeopleController.js [hunk 2]
```diff
@@ -215,7 +215,7 @@
                 }
             });
 
-            modalInstance.result.then(function modalConfirmed() {
+            $scope.modalInstance.result.then(function modalConfirmed() {
```
Comment (elliottyates) [inline course_info/static/course_info/js/controllers/PeopleController.js]: Do we need a `finally()` here that [removes this from the controller scope](https://github.com/Harvard-University-iCommons/canvas_account_admin_tools/blob/task/elliottyates/tlt-2119/course_info_add_multiple_people_jasmine_tests/course_info/static/course_info/js/controllers/PeopleController.js#L252), then, like we do for `$scope.confirmRemoveModalInstance`?

File: course_info/tests/jasmine/controllers/PeopleControllerSpec.js [hunk 3]
```diff
@@ -181,9 +181,64 @@ describe('Unit testing PeopleController', function() {
 
     });
     describe('confirmAddPeopleToCourse', function() {
-        it('reflects the right role and number of people');
-        it('initiates the process if confirmed');
-        it('doesn\'t initiate the process if canceled');
+        beforeEach(function() {
```
Comment (elliottyates) [inline course_info/tests/jasmine/controllers/PeopleControllerSpec.js]: There's now a [one-liner](https://github.com/Harvard-University-iCommons/canvas_account_admin_tools/blob/task/elliottyates/tlt-2119/course_info_add_multiple_people_jasmine_tests/course_info/tests/jasmine/controllers/PeopleControllerSpec.js#L39) that might be able to replace this CI mocking stuff:
```
clearInitialCourseInstanceFetch();
```

File: course_info/tests/jasmine/controllers/PeopleControllerSpec.js [hunk 4]
```diff
@@ -181,9 +181,64 @@ describe('Unit testing PeopleController', function() {
...
+            var ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'confirmAddPeopleToCourse test',
+            };
+            courseInstances.instances[ci.course_instance_id] = ci;
+            controller = $controller('PeopleController', {$scope: scope});
+        });
+
+        xit('reflects the right role and number of people', function() {
```
Comment (elliottyates) [inline course_info/tests/jasmine/controllers/PeopleControllerSpec.js]: I also have no idea how to do this, after some research. Let's just make it pending, and put some comments in it, maybe something like:
```
// todo: cannot yet find a way to access the modal's controller scope within a test
it('reflects the right role and number of people');
```

File: course_info/tests/jasmine/controllers/PeopleControllerSpec.js [hunk 5]
```diff
@@ -181,9 +181,64 @@ describe('Unit testing PeopleController', function() {
...
+            expect(modalScope.selectedRoleName).toEqual("Student");
+        });
+
+        it('initiates the process if confirmed', function() {
+            var searchTerm ="123456789, \n987654321";
+            var searchTermList = [
+                ['123456789', ['123456789']],
+                ['987654321 ', ['987654321']],
+            ];
+            spyOn(scope, 'getSearchTermList').and.returnValue(searchTermList);
```
Comment (elliottyates) [inline course_info/tests/jasmine/controllers/PeopleControllerSpec.js]: I don't think we need to mock this, really. It's a straightforward call that doesn't rely on anything external. In any case, if you wanted to continue to mock it, the `searchTermList` var we'd expect it to return would be:
```
["123456789", "987654321"]
```

File: course_info/tests/jasmine/controllers/PeopleControllerSpec.js [hunk 6]
```diff
@@ -181,9 +181,64 @@ describe('Unit testing PeopleController', function() {
...
+            spyOn(scope, 'addPeopleToCourse');
+            scope.confirmAddPeopleToCourse(searchTerm);
+            scope.$digest();  // resolves modal instantiation
+
+            //The modalInstance in the controller had to be made a scope variable
```
Comment (elliottyates) [inline course_info/tests/jasmine/controllers/PeopleControllerSpec.js]: Since we're going to leave the 'reflects the right role and number of people' test pending, everything above this line can be moved to the `beforeEach()` for these remaining tests.

File: course_info/static/course_info/js/controllers/PeopleController.js (hunk 2 above)
Comment (sapnamysore) [inline course_info/static/course_info/js/controllers/PeopleController.js]: Yes, I had it in my earlier version and put it while applying my stashed changes.

File: course_info/tests/jasmine/controllers/PeopleControllerSpec.js (hunk 3 above)
Comment (sapnamysore) [inline course_info/tests/jasmine/controllers/PeopleControllerSpec.js]: Cool

File: course_info/tests/jasmine/controllers/PeopleControllerSpec.js (hunk 5 above)
Comment (sapnamysore) [inline course_info/tests/jasmine/controllers/PeopleControllerSpec.js]: I hadn't mocked it initially, but it was throwing some errors due to other unrelated stuff that has gone away since then.  And the return was a direct dump from Chrome Dev Tools :-) . Cleaning it
